granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -78,7 +101,94 @@ def find_v2_hierarchy(resolve_host_root_links: bool = True) -> Optional[str]:
     return path
 
 
-def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> Optional[str]:
+CGROUP_PROCS_FILE = "cgroup.procs"
+
+
+class CgroupCore:
+    path: Path
+
+    def __init__(self, path: Path):
+        self.path = path
+
+    def _get_parent_cgroup(self, cgroup_name: str) -> Optional[Path]:
+        if cgroup_name in self.path.parts:
+            return Path(self.path.as_posix().split(cgroup_name)[0]) / cgroup_name
+        return None
+
+    def _get_cgroup_in_hierarchy(self, cgroup_name: str) -> Optional[Path]:
```

Comment:
This function ain't really `Optional`, mypy cannot see through `_get_parent_cgroup` returning non-None the 2nd time you call it, after `has_parent_cgroup` returned true. Instead of using an incorrect type, you should help mypy by adding an `assert new_cgroup_path is not None` after the 2nd `_get_parent_cgroup` call.